sw/device/silicon_creator/rom/e2e/rom_e2e_flash_ctrl_init_test.c

```diff
@@ -0,0 +1,304 @@
+  check_info_page_bank(kFlashCtrlInfoPageBootData1, 1);
+  check_info_page_pagenum(kFlashCtrlInfoPageBootData1, 1);
+
+  // Extract expected values from OTP.
+  uint32_t otp_boot_info_cfg_value =
+      otp_read32(OTP_CTRL_PARAM_CREATOR_SW_CFG_FLASH_INFO_BOOT_DATA_CFG_OFFSET);
+  flash_cfg_reg_t otp_boot_info_cfg = {
+      .reg_value = otp_boot_info_cfg_value,
+      .scrambling_en = FLASH_CTRL_OTP_FIELD_SCRAMBLING,
+      .ecc_en = FLASH_CTRL_OTP_FIELD_ECC,
+      .he_en = FLASH_CTRL_OTP_FIELD_HE,
+  };
+
+  // Check the configuration for bank 1, page 0.
+  uint32_t page0_cfg_value =
+      abs_mmio_read32(kBase + FLASH_CTRL_BANK1_INFO0_PAGE_CFG_0_REG_OFFSET);
+  flash_cfg_reg_t page0_cfg = {
+      .reg_value = page0_cfg_value,
+      .scrambling_en = FLASH_CTRL_BANK1_INFO0_PAGE_CFG_0_SCRAMBLE_EN_0_FIELD,
+      .ecc_en = FLASH_CTRL_BANK1_INFO0_PAGE_CFG_0_ECC_EN_0_FIELD,
+      .he_en = FLASH_CTRL_BANK1_INFO0_PAGE_CFG_0_HE_EN_0_FIELD,
+  };
+  check_cfg_match(page0_cfg, otp_boot_info_cfg);
+
+  // Check the configuration for bank 1, page 1.
+  uint32_t page1_cfg_value =
+      abs_mmio_read32(kBase + FLASH_CTRL_BANK1_INFO0_PAGE_CFG_1_REG_OFFSET);
+  flash_cfg_reg_t page1_cfg = {
+      .reg_value = page1_cfg_value,
+      .scrambling_en = FLASH_CTRL_BANK1_INFO0_PAGE_CFG_1_SCRAMBLE_EN_1_FIELD,
+      .ecc_en = FLASH_CTRL_BANK1_INFO0_PAGE_CFG_1_ECC_EN_1_FIELD,
+      .he_en = FLASH_CTRL_BANK1_INFO0_PAGE_CFG_1_HE_EN_1_FIELD,
+  };
+  check_cfg_match(page1_cfg, otp_boot_info_cfg);
+}
+
+/**
+ * Check that the Creator Secret info page is fully locked down.
+ *
+ * Ensure that:
+ * - The page is not readable, programmable, or erasable.
+ * - The page configuration is not write-enabled.
+ * - An attempt to read from the page returns an error.
+ * - An attempt to write to the page returns an error.
+ */
+static void creator_secret_locked_test(void) {
+  // Double-check the expected bank and page number for the creator secret info
+  // page.  These numbers are hardcoded below to mitigate risk from errors in
+  // macros that construct the page-specific constants.
+  //
+  // Expected values:
+  // * kFlashCtrlInfoPageCreatorSecret -> bank 0, page 1
+  check_info_page_bank(kFlashCtrlInfoPageCreatorSecret, 0);
+  check_info_page_pagenum(kFlashCtrlInfoPageCreatorSecret, 1);
+
+  // Check permission bits.
+  //
+  // Expected values:
+  // * EN        = true (otherwise the configuration does not apply)
+  // * RD_EN     = false
+  // * PROG_EN   = false
+  // * ERASE_EN  = false
+  uint32_t info_page_cfg =
+      abs_mmio_read32(kBase + FLASH_CTRL_BANK0_INFO0_PAGE_CFG_1_REG_OFFSET);
+  // TODO: the below fails because enable is false, but it's unclear from the
+  // flash documentation if this actually disables all the other non-enabled
+  // things. Need to investigate.
+  // CHECK(kMultiBitBool4True ==
+  //      bitfield_field32_read(info_page_cfg,
+  //                            FLASH_CTRL_BANK0_INFO0_PAGE_CFG_1_EN_1_FIELD));
```

Comment:
This is something I wasn't sure about when reading the docs and writing the test. The documentation for `flash_ctrl` says that if the `en` field is set then "the following fields apply": https://docs.opentitan.org/hw/ip/flash_ctrl/doc/#Reg_bank0_info0_page_cfg_0

To me this wording kind of implies that if `en` is unset, the fact that read/prog/erase is disabled in the following fields has no effect and ...some other configuration applies. Is that correct? If so, the ROM driver should probably be modified to set the `en` flag.

@tjaychen for `flash_ctrl` questions, since I see you have a lot of commit history in `hw/ip/flash_ctrl/rtl` :slightly_smiling_face:

Reply:
yeah that's right. At least what you describe is how it should behave. Do let me know if that's not what you see. I'll make the wording more explicit as well.

The `en` concept doesn't make quite as much sense to `info pages`, it just gives you a quick knob to disable all the settings should you need it. It applies a bit more for the `data pages`, because it gives you a way to say I want the controls to fall to the "default'.  The info pages don't really have a default concept right now (in hindsight, maybe I should have added that).

Reply:
No access to the info page if the corresponding `EN` is `false`, right @tjaychen ?

Reply:
Okay, so if everything is disabled, the `en` bit doesn't matter because the "default" in this case is that everything is disabled.  In this case I think we should modify the code anyway to set `en`, just for clarity/in case the info pages do get a default value later on. I can take care of this change if @alphan agrees.

Reply:
@jadephilipoom which code are you referring to? We set those registers in `flash_ctrl_init()`. Please note that these bitfields use mubi variants and they are disabled unless they are set to true.